### PR TITLE
HPCC-16840 Do not log TxSummary if both user and request are empty

### PR DESCRIPTION
**Pull request by wangkx**
Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

### CODE REVIEW

**Comment by wangkx** [comment]
@rpastrana please review.

File: esp/platform/espcontext.cpp [hunk 1]
```diff
@@ -586,12 +586,28 @@ void CEspContext::updateTraceSummaryHeader()
     if (m_txSummary)
     {
         m_txSummary->set("activeReqs", m_active);
-        m_txSummary->set("user", VStringBuffer("%s%s%s", (queryUserId() ? queryUserId() : ""), (m_peer.length() ? "@" : ""), m_peer.str()).str());
+        VStringBuffer user("%s%s%s", (queryUserId() ? queryUserId() : ""), (m_peer.length() ? "@" : ""), m_peer.str());
+        if (!user.isEmpty())
+            m_txSummary->set("user", user.str());
 
-        VStringBuffer reqSummury("%s %s.%s", httpMethod.get(), m_servName.str(), servMethod.get());
+        VStringBuffer reqSummury("%s", httpMethod.isEmpty() ? "" : httpMethod.get());
```
Comment (richardkchapman) [inline esp/platform/espcontext.cpp]: How many ways to misspell summary are there?

**Comment by wangkx** [comment]
Typo has been fixed.

File: esp/platform/espcontext.cpp [hunk 2]
```diff
@@ -586,12 +586,28 @@ void CEspContext::updateTraceSummaryHeader()
     if (m_txSummary)
     {
         m_txSummary->set("activeReqs", m_active);
-        m_txSummary->set("user", VStringBuffer("%s%s%s", (queryUserId() ? queryUserId() : ""), (m_peer.length() ? "@" : ""), m_peer.str()).str());
+        VStringBuffer user("%s%s%s", (queryUserId() ? queryUserId() : ""), (m_peer.length() ? "@" : ""), m_peer.str());
+        if (!user.isEmpty())
+            m_txSummary->set("user", user.str());
 
-        VStringBuffer reqSummury("%s %s.%s", httpMethod.get(), m_servName.str(), servMethod.get());
+        VStringBuffer reqSummary("%s", httpMethod.isEmpty() ? "" : httpMethod.get());
```
Comment (rpastrana) [inline esp/platform/espcontext.cpp]: A couple of questions, Is this code HTTP specific?
Also, what does it meant to have an empty httpmethod?

File: esp/platform/espcontext.cpp [hunk 3]
```diff
@@ -586,12 +586,28 @@ void CEspContext::updateTraceSummaryHeader()
...
+        if (!m_servName.isEmpty() || !servMethod.isEmpty())
+        {
+            if (!reqSummary.isEmpty())
+                reqSummary.append(" ");
+            if (!m_servName.isEmpty())
+                reqSummary.append(m_servName.str());
+            if (!servMethod.isEmpty())
```
Comment (rpastrana) [inline esp/platform/espcontext.cpp]: is it possible servName.isEmpty and !servName.isEmpty?? in that case we'll have reqsummary will contain " .<methodname>"

**Review by rpastrana — COMMENTED** [review]
@wangkx 2 very minor questions

File: esp/platform/espcontext.cpp (hunk 2 above)
Comment (wangkx) [inline esp/platform/espcontext.cpp]: This is not HTTP specific. For example, a 'ping' message have an empty httpmethod.

File: esp/platform/espcontext.cpp (hunk 3 above)
Comment (wangkx) [inline esp/platform/espcontext.cpp]: You must ask: "is it possible servName.isEmpty and !servMethod.isEmpty?" In that case, the reqsummary contains ".servMethod" which indicates a method name (not a service name): servMethod.

**Comment by wangkx** [comment]
@rpastrana please see my answers for your questions.

**Comment by rpastrana** [comment]
@wangkx questions answered. This seem reasonable.

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

**Pull request merged**